HardwareCheckout/terminals.py

```diff
@@ -27,11 +27,13 @@ async def get(self):
 class RWTerminalHandler(UserBaseHandler):
     @authenticated
     async def get(self):
-        if not self.current_user.has_roles("Admin"):
-            self.redirect(self.reverse_url("ROTerminals"))
-            return
-
         with self.make_session() as session:
+            current_user = await as_future(session.query(User).filter_by(id=self.current_user).one)
+
+            if not current_user.has_roles("Admin"):
```

Comment:
What if the user is not an admin, but a normal user requesting a hardware checkout terminal which requires a r/w session?

Reply:
This query lists EVERY terminal that available. We don't provide the RW terminals to the end users through this mechanism. they need to connect using the supplied info that appears at the top of the page. This adds the RW terminals to the bottom of the page so that admins can watch and help out end users.